pkg/scm/git/git.go

```diff
@@ -28,18 +31,50 @@ type Git interface {
 	Clone(source, target string, opts api.CloneConfig) error
 	Checkout(repo, ref string) error
 	SubmoduleUpdate(repo string, init, recursive bool) error
+	LsTree(repo, ref string, recursive bool) ([]os.FileInfo, error)
 	GetInfo(string) *api.SourceInfo
+	CreateEmptyLocalGitDirectory() (string, error)
+	CreateLocalGitDirectory() (string, error)
+	CreateLocalGitDirectoryWithSubmodule() (string, error)
+}
+
+func isUsingCygwinGit() bool {
+	if runtime.GOOS == "windows" {
+		// If we find the cygpath utility (which translates between UNIX-style and
+		// Windows-style paths) in the same directory as git, assume we're using
+		// Cygwin.
+		cygpath, _ := exec.LookPath("cygpath")
+		git, err := exec.LookPath("git")
+		if err == nil && filepath.Dir(cygpath) == filepath.Dir(git) {
```

Comment:
Unlike to happen in practice..., but if I `cd /path/to/git`, then run `s2i`, we might incorrectly return `true` even when `cypath` doesn't exist (because we ignored the error above).